wire/error.go

```diff
@@ -9,6 +9,153 @@ import (
 	"fmt"
 )
 
+// ErrorCode describes a kind of message error.
+type ErrorCode int
+
+// These constants are used to identify a specific Error.
+const (
+	// errDefault is the default error code. It is not intended to be used
+	// explicitly as an error code.
+	errDefault ErrorCode = iota
```

Comment:
A "default error" doesn't make much sense, especially if its purpose is to not use it.  I'd suggest removing this and starting the first error with `iota + 1`, and then just check against a literal 0 elsewhere.

Reply:
noted, thanks.